locations/spiders/gov_osservaprezzi_carburanti_it.py

```diff
@@ -0,0 +1,57 @@
+from scrapy import Spider
+from scrapy.http import JsonRequest, Response
+
+from locations.categories import Categories, Fuel, apply_category, apply_yes_no
+from locations.dict_parser import DictParser
+from locations.geo import point_locations
+
+
+class GovOsservaprezziCarburantiITSpider(Spider):
+    name = "gov_osservaprezzi_carburanti_it"
+
+    FUELS = {
+        "Gasolio": Fuel.DIESEL,
+        "GPL": Fuel.LPG,
+        "Metano": Fuel.CNG,
+        # "Blue Diesel": Fuel.ADBLUE, ?
+    }
+
+    BRANDS = {
+        "Api-Ip": {"brand": "IP", "brand_wikidata": "Q646807"},
+        "Esso": {"brand": "Esso", "brand_wikidata": "Q867662"},
+        "Q8": {"brand": "Q8", "brand_wikidata": "Q1634762"},
+        "Tamoil": {"brand": "Tamoil", "brand_wikidata": "Q706793"},
+        "AgipEni": {"brand": "Eni", "brand_wikidata": "Q565594"},
+        "Giap": {"brand": "Giap", "brand_wikidata": None},
+        "Energas": {"brand": "Energas", "brand_wikidata": None},
+        "Keropetrol": {"brand": "Keropetrol", "brand_wikidata": None},
+        "San Marco Petroli": {"brand": "San Marco Petroli", "brand_wikidata": None},
+    }
+
+    def start_requests(self):
+        for lat, lon in point_locations("italy_grid_10km.csv"):
+            yield JsonRequest(
+                "https://carburanti.mise.gov.it/ospzApi/search/zone",
+                data={
+                    "points": [{"lat": lat, "lng": lon}],
+                    "radius": 10,
+                },
+            )
+
+    def parse(self, response: Response):
+        for result in response.json()["results"]:
+            item = DictParser.parse(result)
+
+            apply_category(Categories.FUEL_STATION, item)
+
+            for fuel_name, fuel in self.FUELS.items():
+                apply_yes_no(fuel, item, any(f["name"] == fuel_name for f in result["fuels"]))
+
+            if (brand := result["brand"]) == "PompeBianche":
+                # non-branded station
+                yield item
+            if brand in self.BRANDS:
+                item.update(self.BRANDS[brand])
+            else:
+                item["brand"] = brand
```

Comment:
`yield` doesn't actually stop the function like `return`, so this specific item continues on to get a brand on line 56. If you want the code to stay like this, you'll need to put a `continue` after the `yield`.

Another approach would be to add `PompeBianche` to `BRANDS`, but without a brand. I find a single "return" less confusing.

```suggestion
            brand = result["brand"]
            if brand in self.BRANDS:
                item.update(self.BRANDS[brand])
            else:
                item["brand"] = brand
                self.crawler.stats.inc_value("atp/gov_osservaprezzi_carburanti_it/unmapped_brand/{}".format(brand))

```

```python
    BRANDS = {
        "Api-Ip": {"brand": "IP", "brand_wikidata": "Q646807"},
        ...
        "PompeBianche": {},  # non-branded station
    }
```

Reply:
Or it could end up in the `if`, something like:

```python
            brand = result["brand"]
            if brand in self.BRANDS:
                item.update(self.BRANDS[brand])
            elif brand == "PompeBianche":
                pass  # non-branded station
            else:
                item["brand"] = brand
                self.crawler.stats.inc_value("atp/gov_osservaprezzi_carburanti_it/unmapped_brand/{}".format(brand))
```

Reply:
I believe that this actually worked somehow, but it was probably because the second "yield" got filtered out as a duplicate. Thanks for the correction.

I think the option with the "empty brand" is the most neat. I'll add the stats as well.